plugins/chat/app/helpers/with_service_helper.rb

```diff
@@ -6,32 +6,21 @@ def result
     end
 
     def with_service(service, default_actions: true, **dependencies, &block)
-      controller = self
+      object = self
       merged_block =
         proc do
-          instance_eval(&controller.default_actions_for_service) if default_actions
-          instance_eval(&(block || proc {}))
+          instance_exec(&object.method(:default_actions_for_service).call) if default_actions
+          instance_exec(&(block || proc {}))
         end
-      Chat::Endpoint.call(service, controller, **dependencies, &merged_block)
+      Chat::Endpoint.call(service, object, **dependencies, &merged_block)
     end
 
     def run_service(service, dependencies)
-      @_result = service.call(params.to_unsafe_h.merge(guardian: guardian, **dependencies.to_h))
+      @_result = service.call(params.to_unsafe_h.merge(guardian: guardian, **dependencies))
     end
 
     def default_actions_for_service
-      proc do
-        on_success { render(json: success_json) }
-        on_failure { render(json: failed_json, status: 422) }
-        on_failed_policy(:invalid_access) { raise Discourse::InvalidAccess }
-        on_failed_contract do
-          render(
-            json:
-              failed_json.merge(errors: result[:"result.contract.default"].errors.full_messages),
-            status: 400,
-          )
-        end
-      end
+      proc {}
```

Comment:
So just to confirm, this does nothing by default if you don't give it any `on_x` methods right?

```ruby
with_service(TestService, **args) do
end
```

At that point you would just call the service directly?

Reply:
Yes, by default there are no default actions. If some pattern emerges from how we’re using jobs, I guess we could add some default actions in the future.
And indeed, your example does nothing :+1: :grin: